picky-asn1-x509/src/pkcs12/encryption/pbes2/pbkdf2.rs

```diff
@@ -0,0 +1,217 @@
+use crate::{oids, RawAlgorithmIdentifier};
+use core::fmt;
+use picky_asn1::Asn1Type;
+use picky_asn1::{
+    tag::TagPeeker,
+    wrapper::{IntegerAsn1, OctetStringAsn1},
+};
+use serde::{de, ser};
+
+/// [PKCS #12: Personal Information Exchange Syntax Standard Version](https://tools.ietf.org/html/rfc7292#appendix-C)
+/// # Appendix C. Keys and IVs for Password Privacy Mode
+/// ```not_rust
+/// PBKDF2-params ::= SEQUENCE {
+///     salt CHOICE {
+///       specified OCTET STRING,
+///       otherSource AlgorithmIdentifier {{PBKDF2-SaltSources}}
+///     },
+///     iterationCount INTEGER (1..MAX),
+///     keyLength INTEGER (1..MAX) OPTIONAL,
+///     prf AlgorithmIdentifier {{PBKDF2-PRFs}} DEFAULT
+///     algid-hmacWithSHA1
+/// }
+/// ```
+///
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Pbkdf2Params {
+    pub salt: Pbkdf2SaltSource,
+    pub iteration_count: u32,
+    pub key_length: Option<u32>,
+    /// Defaults to HMAC-SHA1 if absent
+    pub prf: Option<Pbkdf2Prf>,
+}
+
+impl ser::Serialize for Pbkdf2Params {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: ser::Serializer,
+    {
+        use ser::SerializeSeq;
+
+        let sequence_size = 2 + self.key_length.is_some() as usize + self.prf.is_some() as usize;
```

Comment:
Nit: speaking of which! Here I simply suggest we use `usize::from` instead of the `as` as mentioned above. If you decide to address this nit, can you do the same at all the other places? Thank you!